### PR TITLE
Ports tgstation/-tg-station#16569, World Topic() changes

### PR DESCRIPTION
**Pull request by tigercat2000**
tgstation/-tg-station#16569
Full changes:
 - World Topic() is now standardized to use if("parameter" in topic_list)
   instead of an amalgamation of findtext() and if(topic == "parameter").
 - Comms password is now verified at the top of world/Topic() and sets a
   variable to true if it is present and correct.
 - "status" now returns extra information if the comms_password is
   provided and correct.

Misc changes:
 - Shuttle text has been standardized as a proc on mobile docking ports.
   This is used by status/supply displays and mob/Stat() for the -ETA- etc stuff.]
 - Adds another exception to the .gitignore, `.atom-build.json`. It's a file used by the `build` plugin for atom, and it has to be setup per person, so.

### CODE REVIEW

File: code/game/machinery/status_display.dm [hunk 1]
```diff
@@ -88,25 +88,18 @@
...
-						message1 = "-ETR-"
-					if(SHUTTLE_CALL)
-						message1 = "-ETA-"
-					if(SHUTTLE_DOCKED)
-						message1 = "-ETD-"
-					else
-						message1 = "-ERR-"
-						message2 = "??:??"
+				use_warn = 1
+				message1 = "-[shuttle_master.emergency.getModeStr()]"
```
Comment (Tastyfish) [inline code/game/machinery/status_display.dm]: You missed the "-" at the end of the string.

**Comment by Tastyfish** [comment]
I like that you ported the two completely unrelated things that the original PR did. :P

**Comment by tigercat2000** [comment]
@Tastyfish diffs2hard
and you know, something something.. something.

**Comment by Core0verload** [comment]
They were related until I decided to use machine-readable shuttle status and shuttle timer instead of human-readable one.

**Pull request merged**